### PR TITLE
Added user group management

### PR DESCRIPTION
**Pull request by ajcrowe**
This added support for automatically adding users to the docker group to allow access to the daemon.

### CODE REVIEW

**Comment by garethr** [comment]
Hi @ajcrowe. Thanks for this, looks useful. Would you be able to add some simple unit tests to verify the functinality. It really helps with making sure future contributions don't break the functionality. The [Contributing](https://github.com/garethr/garethr-docker/blob/master/CONTRIBUTING.md) doc has some details but do please ask for assistance if required.

**Comment by ajcrowe** [comment]
Sure @garethr I'll get some added

**Comment by ajcrowe** [comment]
Ok i've added some spec and acceptance tests, let me know if you'd like anything else changing

File: spec/defines/system_user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    let(:params) { {'create_user' => true} }
+    it { should contain_user('testuser') }
+    it { should contain_exec('docker-system-user-testuser').with_command(/docker testuser/) }
+    it { should contain_exec('docker-system-user-testuser').with_unless(/grep -qw testuser/) }
+  end
+
+  context 'with create_user => false' do
+    let(:params) { {'create_user' => false} }
+    it { should contain_exec('docker-system-user-testuser').with_command(/docker testuser/) }
+  endg
```
Comment (garethr) [inline spec/defines/system_user_spec.rb]: typo, `endg` instead of `end`. I presume this is what caused the Travis build to fail.

**Comment by garethr** [comment]
Thanks @ajcrowe. Much appreciated. Could you rebase against master so it cleanly merges and I'll look to get this merged in. Cheers

**Comment by ajcrowe** [comment]
I've rebased and squashed the commits, let me know if this is ok

**Comment by garethr** [comment]
@ajcrowe thanks for this, merging now.

**Pull request merged**